Hoc patch in tres partes divisa est.

The change to common.sh fixes a bug where, if VERBOSE is not set, all the
tests fail because "set -u" is called in the testrunner.

The changes to all the flip tests modify them to use bash's build-in echo
because /bin/echo on Solaris doesn't understand command-line flags.

The rest of the patches change instances of "diff" and "patch" to use the
corresponding environment variables like the rest of the tests do.  This is
necessary because diff and patch are expected to be the GNU versions.

All three changes are suitable for pushing upstream.

Note that all the run-test scripts are patched by the makefile to use bash
explicitly.  This change could be upstreamed as well.

--- patchutils-0.3.4/tests/common.sh.~1~	2015-04-14 09:32:02.000000000 -0700
+++ patchutils-0.3.4/tests/common.sh	2017-06-13 19:01:43.124975598 -0700
@@ -1,4 +1,4 @@
-[ -n "$VERBOSE" ] && set -x
+[ -n "${VERBOSE:-}" ] && set -x
 rm -rf ${ARENA:=test-arena}
 mkdir -p $ARENA
 top_builddir=`cd ${top_builddir-.}; pwd`

--- patchutils-0.3.1/tests/flip1/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip1/run-test	Fri Dec 18 10:36:24 2015
@@ -29,7 +29,7 @@
 ${DIFF} -u file.orig file > patch2
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file.orig
 +++ file
--- patchutils-0.3.1/tests/flip10/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip10/run-test	Fri Dec 18 10:36:24 2015
@@ -27,7 +27,7 @@
 EOF
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file.orig
 +++ file
--- patchutils-0.3.1/tests/flip11/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip11/run-test	Fri Dec 18 10:36:24 2015
@@ -27,7 +27,7 @@
 EOF
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file.orig
 +++ file
--- patchutils-0.3.1/tests/flip12/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip12/run-test	Fri Dec 18 10:36:24 2015
@@ -79,7 +79,7 @@
 ${DIFF} -u file.orig file > patch2
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file.orig
 +++ file
--- patchutils-0.3.1/tests/flip13/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip13/run-test	Fri Dec 18 10:36:24 2015
@@ -87,7 +87,7 @@
 ${DIFF} -u file.orig file > patch2
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file.orig
 +++ file
--- patchutils-0.3.1/tests/flip14/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip14/run-test	Fri Dec 18 10:36:24 2015
@@ -32,7 +32,7 @@
 ${DIFF} -u file.orig file > patch2
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file.orig
 +++ file
--- patchutils-0.3.1/tests/flip15/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip15/run-test	Fri Dec 18 10:36:24 2015
@@ -27,7 +27,7 @@
 ${DIFF} -u file.orig file > patch2
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file.orig
 +++ file
--- patchutils-0.3.1/tests/flip16/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip16/run-test	Fri Dec 18 10:36:24 2015
@@ -469,7 +469,7 @@
 ${DIFF} -u file.orig file > patch2
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file.orig
 +++ file
--- patchutils-0.3.1/tests/flip2/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip2/run-test	Fri Dec 18 10:36:24 2015
@@ -29,7 +29,7 @@
 ${DIFF} -u file.orig file > patch2
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file.orig
 +++ file
--- patchutils-0.3.1/tests/flip3/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip3/run-test	Fri Dec 18 10:36:24 2015
@@ -29,7 +29,7 @@
 ${DIFF} -u file.orig file > patch2
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file.orig
 +++ file
--- patchutils-0.3.1/tests/flip4/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip4/run-test	Fri Dec 18 10:36:24 2015
@@ -53,7 +53,7 @@
 ${DIFF} -u file.orig file > patch2
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file.orig
 +++ file
--- patchutils-0.3.1/tests/flip5/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip5/run-test	Fri Dec 18 10:36:24 2015
@@ -59,7 +59,7 @@
 ${DIFF} -u file.orig file > patch2
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file.orig
 +++ file
--- patchutils-0.3.1/tests/flip6/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip6/run-test	Fri Dec 18 10:36:24 2015
@@ -63,7 +63,7 @@
 ${DIFF} -u file.orig file > patch2
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file.orig
 +++ file
--- patchutils-0.3.1/tests/flip7/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip7/run-test	Fri Dec 18 10:36:24 2015
@@ -63,7 +63,7 @@
 ${DIFF} -u file.orig file > patch2
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file.orig
 +++ file
--- patchutils-0.3.1/tests/flip8/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip8/run-test	Fri Dec 18 10:36:24 2015
@@ -109,7 +109,7 @@
 ${DIFF} -u file.orig file > patch2
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file.orig
 +++ file
--- patchutils-0.3.1/tests/flip9/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/flip9/run-test	Fri Dec 18 10:36:24 2015
@@ -25,7 +25,7 @@
 ${DIFF} -u file2.orig file2 > patch2
 
 ${FLIPDIFF} patch1 patch2 > patch-flipped || exit 1
-sed -e "s/$(/bin/echo -ne '\t').*$//" patch-flipped > patch-cmp
+sed -e "s/$(echo -ne '\t').*$//" patch-flipped > patch-cmp
 cmp - patch-cmp << EOF || exit 1
 --- file2.orig
 +++ file2

--- patchutils-0.3.1/tests/combine1/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/combine1/run-test	Fri Dec 18 10:36:24 2015
@@ -96,6 +96,6 @@
 [ -s errors ] && exit 1
 
 # There should be no fuzz.
-patch file.a-z < patches >fuzz || exit 1
-[ -s errors ] && exit 1
+${PATCH} file.a-z < patches >fuzz || exit 1
+
 exit 0
--- patchutils-0.3.1/tests/nodate/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/nodate/run-test	Fri Dec 18 10:36:24 2015
@@ -20,7 +20,7 @@
 ${INTERDIFF} -U0 patch1 patch2 2>errors > patch1-2 || exit 1
 [ -s errors ] && exit 1
 cat << EOF | cmp - patch1-2 || exit 1
-diff -U0 file file
+${DIFF} -U0 file file
 --- file
 +++ file	Nov 27 2001
 @@ -36 +36 @@
--- patchutils-0.3.1/tests/unline1/run-test.~1~	Wed Jul  2 04:11:08 2008
+++ patchutils-0.3.1/tests/unline1/run-test	Fri Dec 18 10:36:24 2015
@@ -25,7 +25,7 @@
 [ -s errors ] && exit 1
 
 cmp - patch1-2 << EOF || exit 1
-diff -u file file
+${DIFF} -u file file
 --- file
 +++ file
 @@ -1 +1 @@
--- patchutils-0.3.4/tests/comma/run-test.orig	2017-06-13 15:53:04.998330685 -0700
+++ patchutils-0.3.4/tests/comma/run-test	2017-06-13 15:58:29.475915138 -0700
@@ -25,7 +25,7 @@
 ( for f in file xyzzy
   do
     printf "diff -up dir/$f.orig dir/$f\n"
-    diff -up dir/$f.orig dir/$f
+    ${DIFF} -up dir/$f.orig dir/$f
   done) > patch1
 
 ${INTERDIFF} patch1 patch1 2>errors > patch1-2 || exit 1
--- patchutils-0.3.4/tests/gendiff1/run-test.orig	2017-06-13 15:53:04.994773049 -0700
+++ patchutils-0.3.4/tests/gendiff1/run-test	2017-06-13 15:57:53.102279156 -0700
@@ -20,11 +20,11 @@
 cp abc dir/file
 # "gendiff dir .orig" without requiring gendiff:
 (printf "diff -up dir/file.orig dir/file\n"
- diff -up dir/file.orig dir/file) > patch1
+ ${DIFF} -up dir/file.orig dir/file) > patch1
 
 cp abd dir/file
 (printf "diff -up dir/file.orig dir/file\n"
- diff -up dir/file.orig dir/file) > patch2
+ ${DIFF} -up dir/file.orig dir/file) > patch2
 
 ${INTERDIFF} patch1 patch2 2>errors > patch1-2 || exit 1
 [ -s errors ] && exit 1
--- patchutils-0.3.4/tests/gendiff2/run-test.orig	2017-06-13 15:53:05.012620642 -0700
+++ patchutils-0.3.4/tests/gendiff2/run-test	2017-06-13 15:58:09.499824312 -0700
@@ -20,12 +20,12 @@
 cp abc dir/file
 # "gendiff dir .orig" without requiring gendiff:
 (printf "diff -up dir/file.orig dir/file\n"
- diff -up dir/file.orig dir/file) > patch1
+ ${DIFF} -up dir/file.orig dir/file) > patch1
 
 cp abd dir/file
 mv dir/file.orig dir/file.old
 (printf "diff -up dir/file.old dir/file\n"
- diff -up dir/file.old dir/file) > patch2
+ ${DIFF} -up dir/file.old dir/file) > patch2
 
 ${INTERDIFF} patch1 patch2 2>errors > patch1-2 || exit 1
 [ -s errors ] && exit 1
